src/simulator_core/solver/network/assets/node.py

```diff
@@ -96,9 +108,15 @@ def get_equations(self) -> list[EquationObject]:
             A list of EquationObjects that contain the indices, coefficients, and right-hand side
             values of the equations.
         """
-        equations = [self.add_node_cont_equation(),
-                     self.add_energy_equations(),
-                     self.add_discharge_equation()]
+        # Check connection
+        if not self.is_connected():
+            raise ValueError(f"Node {self.name} is not connected to any asset.")
+        # Construct equation object
+        equations = [
+            self.add_node_cont_equation(),
+            self.add_energy_equations(),
+            self.add_discharge_equation(),
+        ]
         return equations
 
     def add_energy_equations(self) -> EquationObject:
```

Comment:
Dit blijft een grote method, misschien goed om een issue aan te maken om deze op te splitsen in kleinere stukken

Reply:
Zie issue #81 to construct a method to get equation objects